records/generator.go

```diff
@@ -293,27 +296,32 @@ func (rg *RecordGenerator) InsertState(sj StateJSON, domain string, ns string,
 			tag := hashString(task.ID)
 			tail := fname + "." + domain + "."
 
-			// A records for task and task-sid
 			arec := tname + "." + tail
-			rg.insertRR(arec, hostIP, "A")
 			trec := tname + "-" + tag + "-" + sid + "." + tail
-			rg.insertRR(trec, hostIP, "A")
 
-			// A records with container IP
-			if containerIP := task.containerIP(); containerIP != "" {
-				rg.insertRR("_container."+arec, containerIP, "A")
-				rg.insertRR("_container."+trec, containerIP, "A")
+			containerIP := ""
+			if containerIPsOn {
+				containerIP = task.containerIP()
 			}
-
-			// SRV records
-			if task.Resources.Ports != "" {
-				ports := yankPorts(task.Resources.Ports)
-				for _, port := range ports {
-					srvhost := trec + ":" + port
-					tcp := "_" + tname + "._tcp." + tail
-					udp := "_" + tname + "._udp." + tail
-					rg.insertRR(tcp, srvhost, "SRV")
-					rg.insertRR(udp, srvhost, "SRV")
+			if containerIP != "" {
+				// A records with container IP, no SRV records here
+				rg.insertRR(arec, containerIP, "A")
+				rg.insertRR(trec, containerIP, "A")
+			} else {
```

Comment:
to clarify, those SRV's would look like this:
`_my-task._tcp.my-framework.mesos` --> `my-task.my-framework.slave.mesos:{host-port}`

this should provide backwards compat for existing clients that are relying on SRV's generated for tasks that have allocated host ports.